cmd/fpga_plugin/mode.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getModeOverrideFromCluster(kubeconfig, master, mode string) (string, error) {
+	var config *rest.Config
+	var err error
+
+	if kubeconfig == "" {
+		config, err = rest.InClusterConfig()
+	} else {
+		config, err = clientcmd.BuildConfigFromFlags(master, kubeconfig)
+	}
+	if err != nil {
+		return mode, err
+	}
+
+	nodeName := os.Getenv("NODE_NAME")
```

Comment:
Can we also have a command line option --node-name with the default value = os.Hostname() ?

Reply:
the parameters to this function is already quite a lot. I've not used the plugin so I'm not in a position to comment whether the option is useful or not but my intuition says having the controls via `NODE_NAME` is enough.

what other think?

Reply:
Basically there two ways to run this plugin
1. inside a pod and
2. on the host.
In the first case `NODE_NAME` is always set and it'd be an error otherwise. Therefore having the possibility to override with an option is prone to admin mistakes. To me this sounds like an UX issue.

Reply:
If the use case is get the mode setting from a certain node, the same can be achieved by setting the name as the environment in the deployment (instead of `spec.nodeName`), right?

Reply:
(my comment was added without seeing @rojkov comment first)

Reply:
That's right. Cloud native apps are supposed to be configured with env variables only (well, that's a recommendation). 

What Ed is talking about is developer experience. It may be convenient to automatically detect the correct node's name when developing the plugin, but IMHO this adds magic and makes less obvious to figure out what's wrong once hostname and the node's name (working on the same host) are different. For this reason I'd remove the `os.Hostname()` part completely.

Personally as a developer I find `NODE_NAME=host1 ./fpga_plugin` and `./fpga_plugin --node-name=host1` equally convenient.

Reply:
I found ./fpga_plugin --node-name=<name> with name=os.Getenv("NODE_NAME") by default much more explicit and logical.

Reply:
Allright. Sounds good to me now.

Reply:
added. PTAL.